src/components/QuranReader/index.tsx

```diff
@@ -91,6 +92,16 @@ const QuranReader = ({
     QURAN_READER_OBSERVER_ID,
   );
 
+  const lines = useMemo(
```

Comment:
There is an issue with this is that:

1. For `ReadingView`, the resource-intensive function of `groupLinesByVerses` will be executed at least twice. One time here and another inside `src/components/QuranReader/ReadingView/Page.tsx`.
2. For `TranslationView`, this adds extra calculations and won't be used since `QuranReader` component is shared between both views.